krux/io.py

```diff
@@ -89,17 +90,28 @@ def is_regex(thing):
         if shell is None:
             shell = isinstance(command, basestring)
 
+        # GOTCHA: If the process is created using shell, upon timeout, the shell process will be
+        # terminated but not the actual command. Use 'exec' shell keyword so that the actual command's process
+        # is terminated and prevent a false timeout.
+        if shell is True and timeout is not None:
+            command = 'exec ' + command
```

Comment:
Yeah, I am torn about this as well. It is nice not to worry about this, but at the same time, this can really mess up the command.